packages/terra-profile-image/src/ProfileImage.jsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import 'terra-base/lib/baseStyles';
+import TerraImage from 'terra-image';
+import avatarImage from './avatar.svg';
+import './ProfileImage.scss';
+
+const propTypes = {
+  /**
+   * The profile image to load.
+   */
+  image: PropTypes.element.isRequired,
+};
+
+class ProfileImage extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { isLoading: true, isError: false };
+
+    this.updateImage(props.image.props);
+  }
+
+  componentWillReceiveProps(newProps) {
+    this.updateImage(newProps.image.props);
+  }
+
+  updateImage(imageProps) {
+    const avatarOverrideProps = { src: avatarImage, onLoad: undefined, onError: undefined };
```

Comment:
This `updateImage` has been called in `componentWillReceiveProps`, so it will be called every time if there is a props update to rerender the component. In that case, for each rerender, you will create a new object for hold the `avatorOverrideProps`. If we put such constant object outside of the Class, it will just create once. And pass the reference for each rerender to save the performance.